### ISSUE TITLE
`*args` support is broken: only strings are supported as `*args`

### ISSUE THREAD

**Issue by AIGeneratedUsername** [Krukov/cashews#65]
I've been using a fork (3.x "cashews" version), but today I've decided to finally switch to your latest version 4.2.1. After switching I see that some problems were introduced somewhere after the 3.x version.

Here is a reproducible code sample: 

```python
import asyncio

from cashews import cache


@cache(ttl="1s")
async def get_name(user, *args, version="v1", **kwargs):
    ...


asyncio.run(get_name("foo", 999.0, spam="eggs"))
```

Traceback:

```
Traceback (most recent call last):
  File "/demo/_local.py", line 12, in <module>
    asyncio.run(get_name("foo", 999.0, spam="eggs"))
  File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/demo-vUAg5yd0-py3.10/lib/python3.10/site-packages/cashews/wrapper.py", line 248, in _call
    return await decorator(*args, **kwargs)
  File "/demo-vUAg5yd0-py3.10/lib/python3.10/site-packages/cashews/decorators/cache/simple.py", line 37, in _wrap
    _cache_key = get_cache_key(func, _key_template, args, kwargs)
  File "/demo-vUAg5yd0-py3.10/lib/python3.10/site-packages/cashews/key.py", line 69, in get_cache_key
    return template_to_pattern(_key_template, _formatter=default_formatter, **key_values)
  File "/demo-vUAg5yd0-py3.10/lib/python3.10/site-packages/cashews/formatter.py", line 124, in template_to_pattern
    return _formatter.format(template, **values)
  File "/usr/lib/python3.10/string.py", line 161, in format
    return self.vformat(format_string, args, kwargs)
  File "/usr/lib/python3.10/string.py", line 165, in vformat
    result, _ = self._vformat(format_string, args, kwargs, used_args, 2)
  File "/usr/lib/python3.10/string.py", line 218, in _vformat
    result.append(self.format_field(obj, format_spec))
  File "/demo-vUAg5yd0-py3.10/lib/python3.10/site-packages/cashews/formatter.py", line 94, in format_field
    value = super().format_field(value, format_spec if format_spec not in self._functions else "")
  File "/demo-vUAg5yd0-py3.10/lib/python3.10/site-packages/cashews/formatter.py", line 72, in format_field
    return format(self._format_field(value))
  File "/demo-vUAg5yd0-py3.10/lib/python3.10/site-packages/cashews/formatter.py", line 65, in _format_field
    return self.__type_format[_type](value)
  File "/demo-vUAg5yd0-py3.10/lib/python3.10/site-packages/cashews/formatter.py", line 18, in _decode_array
    return ":".join([format_value(value) for value in values])
TypeError: sequence item 0: expected str instance, float found

Process finished with exit code 1
```

To reproduce the error you can pass any non-string value.

Also, this bug shows a lack of test coverage for the new code.

**Comment by Krukov** [comment]
Thanks for reporting, yep my bad with lack of coverage and bad test cases. I'll try to fix it ASAP

**Comment by Krukov** [comment]
Fixed with 4.2.2 release - please confirm it

**Comment by AIGeneratedUsername** [comment]
Thank you. Now I see no issues.
